core/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -204,10 +204,13 @@ public String getLocalFilename(int blockIndex) throws IOException {
    * @throws IOException
    */
   public List<String> getLocationHosts() throws IOException {
+    List<String> ret = new ArrayList<String>();
```

Comment:
Makes sense, I think we have an early return because calling getClientBlockInfo(0) will throw an exception.